player/lua/utils.lua

```diff
@@ -0,0 +1,82 @@
+--[[
+This file is part of mpv.
+
+mpv is free software; you can redistribute it and/or
+modify it under the terms of the GNU Lesser General Public
+License as published by the Free Software Foundation; either
+version 2.1 of the License, or (at your option) any later version.
+
+mpv is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public
+License along with mpv.  If not, see <http://www.gnu.org/licenses/>.
+]]
+
+local utils = require "mp.utils"
+
+local default_config_file = [[# https://mpv.io/manual/master/#configuration-files
+# https://mpv.io/manual/master/#options
+
+]]
+
+local default_input_conf = [[# https://mpv.io/manual/master/#command-interface
+
+]]
+
+local function show_error(message)
+    mp.msg.error(message)
+    if mp.get_property_native("vo-configured") then
+        mp.osd_message(message)
+    end
+end
+
+local function edit_config_file(filename, initial_contents)
+    local path = mp.find_config_file(filename)
+
+    if not path then
+        path = mp.command_native({"expand-path", "~~/" .. filename})
+        local file_handle, error_message = io.open(path, "w")
+
+        if not file_handle then
+            show_error(error_message)
+            return
+        end
+
+        file_handle:write(initial_contents .. "\n")
+        file_handle:close()
+    end
+
+    local platform = mp.get_property("platform")
+    local args
+    if platform == "windows" then
+        args = {"powershell", "/C", "start", path}
```

Comment:
just run explorer, this works the same as xdg-open. Needs normalized path though, but I would even suggest to always normalize before passing to subproject also on other platforms.

```suggestion
        args = {"explorer", mp.command_native({"normalize-path", path})}
```

Reply:
With `explorer` you may potentially run into [this](https://github.com/tomasklaen/uosc/pull/800) issue if someone uses a portable config and ends up with a path like `C:\apps\,inPathWithoutSpaces\mpv\portable_config`.

Reply:
Copied your fix.